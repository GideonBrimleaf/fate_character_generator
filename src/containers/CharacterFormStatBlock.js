import React from 'react'
import './CharacterForm.css'
import '../components/CharacterDetail.css'

const CharacterFormStatBlock = (props) => {

  const statNames = Object.keys(props.stats)
  const prettyTableName = props.tableName.charAt(0).toUpperCase() + props.tableName.slice(1)

  const tableRows = statNames.map(stat => {
    const capitalisedFirstLetter = stat.charAt(0).toUpperCase() + stat.slice(1)
    const splitName = capitalisedFirstLetter.split(/(?=[A-Z])/)
    const prettyStatLabel = splitName.join(' ') 

    return (
      <tr key={stat}>
        <td className="stat-name">
          <label htmlFor={stat}>{ prettyStatLabel }:</label>
        </td>
        <td className="form-input">
          <input
            className={`character-sheet-text ${props.statBlockType}-input`}
            type="text" 
            id={stat}
            value={props.stats[stat]}
            onChange={(event) => props.handleFormStatChange(event, props.tableName)}
            required
          />
        </td>
      </tr>
    )
  })

  return(
    <table>
      <caption>{ prettyTableName }</caption>
      <tbody>
        { tableRows }
      </tbody>
    </table>
  )
}

export default CharacterFormStatBlock