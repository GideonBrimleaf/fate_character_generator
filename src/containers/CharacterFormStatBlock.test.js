/* eslint-disable react/jsx-props-no-spreading */

import React from 'react';
import { render } from '@testing-library/react';

import { createFakeCharacter } from '../test/fakes/character.fake';
import CharacterFormStatBlock from './CharacterFormStatBlock';

it('renders correctly', () => {
  // Arrange
  const character = createFakeCharacter(1234);

  const props = {
    stats: character.consequences,
    tableName: 'expected table name',
  }

  // Act
  const result = render(
    <CharacterFormStatBlock {...props} />,
  );

  // Assert
  expect(result.asFragment()).toMatchSnapshot();
});
