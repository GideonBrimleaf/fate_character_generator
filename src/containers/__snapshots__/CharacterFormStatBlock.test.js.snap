// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <table>
    <caption>
      Expected table name
    </caption>
    <tbody>
      <tr>
        <td
          class="stat-name"
        >
          <label
            for="mild"
          >
            Mild:
          </label>
        </td>
        <td
          class="form-input"
        >
          <input
            class="character-sheet-text undefined-input"
            id="mild"
            type="text"
            value="1234 mild"
          />
        </td>
      </tr>
      <tr>
        <td
          class="stat-name"
        >
          <label
            for="moderate"
          >
            Moderate:
          </label>
        </td>
        <td
          class="form-input"
        >
          <input
            class="character-sheet-text undefined-input"
            id="moderate"
            type="text"
            value="1234 moderate"
          />
        </td>
      </tr>
      <tr>
        <td
          class="stat-name"
        >
          <label
            for="severe"
          >
            Severe:
          </label>
        </td>
        <td
          class="form-input"
        >
          <input
            class="character-sheet-text undefined-input"
            id="severe"
            type="text"
            value="1234 severe"
          />
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
